app/src/ui/changes/filter-changes-list.tsx

```diff
@@ -859,6 +861,12 @@ export class FilterChangesList extends React.Component<
       this.props.repository.gitHubRepository === null ||
       hasWritePermission(this.props.repository.gitHubRepository)
 
+    const allFilesToCommitNotVisible =
+      this.props.askForConfirmationOnCommitFilteredChanges &&
+      filesSelected.some(
+        file => !this.state.filteredItems.find(fi => fi.id === file.id)
+      )
```

Comment:
I'm a little bit concerned about this from a performance standpoint. At a minimum we should implement a happy path here. If we started by checking if the number of files selected equal the number of filtered items and nope out if that's the case then at least we're not adding any cost if the list isn't filtered.

On the one hand we typically see few files in changes but on the other hand the users that for whatever reason have the most amount of changed files are likely to be the biggest users of filtered changes so I am inclined to say we should make sure this performs well in that scenario.

Creating a set from filteredItems should to the trick

Reply:
Done.

> minimum we should implement a happy path here. If we started by checking if the number of files selected equal the number of filtered items

In this case, the happy path is SelectedItems.length > filteredItems.length, because if filtered items is [1, 2, 3] and selectedItems is [1, 2] or [1, 2, 5], the lengths could be equal or not equal but all the selected items are visible.